### PR TITLE
Fix new module

### PR DESCRIPTION
**Pull request by tomersolomon1**
- Migrate the latest version of the dremio module to work with the version 1.1 of Fybrik
- Refactoring

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,71 +8,103 @@ Ensure that you have the following:
...
+### Install Dremio (Optional)
+You can install a Dremio cluster using the following command:
 ```bash
-kubectl apply -f dremio-module.yaml -n fybrik-system
+helm install <chart-name> charts/dremio-cluster
 ```
 
-### Create namespace
+### Register the Fybrik module:
+In `dremio-module.yaml` can you specify the host and port of an already existing the Dremio cluster. If Dremio is deployed using the previous step, you can set the dremio parameters in `dremio-module.yaml` as the following:
```
Comment (Mohammad-nassar10) [inline README.md:20]: an already existing Dremio cluster

File: README.md [hunk 2]
```diff
@@ -8,71 +8,103 @@ Ensure that you have the following:
...
+
+### Create namespace
+```bash
+kubectl create namespace fybrik-sample
+kubectl config set-context --current --namespace=fybrik-sample
+```
 
-### Register iceberg asset
-Replace the values of `endpoint`, `bucket`, and `object_key` in `sample_asset/asset-iceberg.yaml` file according to your created asset. Then, add the asset to the internal catalog using the following command:
+### Register Icberg asset
```
Comment (Mohammad-nassar10) [inline README.md:47]: Iceberg

File: Pipfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+fybrik_python_logging = "==0.1.0"
+fybrik_python_vault = "==0.1.0"
```
Comment (roytman) [inline Pipfile:8]: the latest python_vault version is 0.2.0

File: README.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# DremioModule
+# Dremio Module
```
Comment (roytman) [inline README.md:2]: maybe make sense to add a sentence or two about what is the DremioModules, why do we need it.

File: charts/dremio-cluster/Chart.yaml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
-name: dremio-module-chart
+name: dremio-chart
 description: A Helm chart for installing a module for Fybrik in Kubernetes
```
Comment (roytman) [inline charts/dremio-cluster/Chart.yaml:3]: A Helm chart for installing a **Dremio** module for Fybrik in Kubernetes

File: charts/dremio-module/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v2
+name: dremio-module-multi-chart
+description: A Helm chart for installing a module for Fybrik in Kubernetes
```
Comment (roytman) [inline charts/dremio-module/Chart.yaml:3]: A Helm chart for installing a **Dremio** module for Fybrik in Kubernetes

File: charts/dremio-module/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: "0.0.0"
+dependencies:
+- name: dremio-chart
+  version: "0.0.0"
+  repository: "file://../dremio-chart"
```
Comment (roytman) [inline charts/dremio-module/Chart.yaml:20]: we don't have dremio-chart now

File: charts/dremio-module/values.yaml [hunk 8]
```diff
@@ -0,0 +1,90 @@
+# Default values for dremio-module.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+imageChart: 
+  repository: ghcr.io/fybrik/dremio-module-register
+  tag: main
+  pullPolicy: Always
```
Comment (roytman) [inline charts/dremio-module/values.yaml:8]: at some places you use `Always` and at others `IfNotPresent`

File: sample/fybrikapplication.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+apiVersion: app.fybrik.io/v1beta1
```
Comment (roytman) [inline sample/fybrikapplication.yaml:1]: Should it be part of the sample ?

File: sample/secret-iceberg.yaml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: iceberg-dataset
+type: Opaque
+stringData:
+  access_key: "b11fbad42c714605a1bfaf65e9a304e0"
```
Comment (roytman) [inline sample/secret-iceberg.yaml:7]: do you really want to store your access and secret keys ?

**Review by roytman — COMMENTED** [review]
why do we store binary (charts/dremio-module/charts/dremio-chart-0.0.0.tgz) file ?

File: sample/secret-iceberg.yaml [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+  secret_key: "7e34abcb1a8bb3a2e77ab62b382f681d6afc9a017feeb629"
```
Comment (Mohammad-nassar10) [inline sample/secret-iceberg.yaml:8]: I think you should remove these keys.

File: charts/dremio-module/Chart.yaml (hunk 7 above)
Comment (Mohammad-nassar10) [reply charts/dremio-module/Chart.yaml:20]: The `dremio-cluster` chart has the name `dremio-chart` (in the file `charts/dremio-cluster/Chart.yaml`). Maybe it is better to change the name of the chart to `dremio-cluster`?

File: sample/fybrikapplication.yaml (hunk 9 above)
Comment (tomersolomon1) [reply sample/fybrikapplication.yaml:1]: Yes, it's a sample fybrik application isn't it?

File: README.md (hunk 4 above)
Comment (tomersolomon1) [reply README.md:2]: Added a sentence or two.

File: README.md [hunk 12]
```diff
@@ -8,71 +10,107 @@ Ensure that you have the following:
...
+Either way, apply the fybrik module using the following command:
 ```bash
-kubectl create namespace fybrik-notebook-sample
-kubectl config set-context --current --namespace=fybrik-notebook-sample
+kubectl apply -f dremio-module.yaml -n fybrik-system
 ```
 
 ### Create iceberg asset
-To be done.
+TBD
```
[the remark this answers is not in the document]
Comment (tomersolomon1) [reply README.md:41]: Done

File: README.md (hunk 1 above)
Comment (tomersolomon1) [reply README.md]: Done

File: Pipfile (hunk 3 above)
Comment (tomersolomon1) [reply Pipfile]: Done

File: README.md (hunk 4 above)
Comment (tomersolomon1) [reply README.md:2]: See current brief explanation

File: charts/dremio-cluster/Chart.yaml (hunk 5 above)
Comment (tomersolomon1) [reply charts/dremio-cluster/Chart.yaml]: Fixed

File: charts/dremio-module/Chart.yaml (hunk 6 above)
Comment (tomersolomon1) [reply charts/dremio-module/Chart.yaml]: Fixed

File: charts/dremio-module/Chart.yaml (hunk 7 above)
Comment (tomersolomon1) [reply charts/dremio-module/Chart.yaml]: Changed the structure + names

**Review by roytman — COMMENTED** [review]
do we have to store in the repo the python cache and especially the `module/__pycache__/utils.cpython-36.pyc` binary file?

**Comment by tomersolomon1** [comment]
@roytman Removed the cache

**Review by roytman — APPROVED** [review]

**Pull request merged**